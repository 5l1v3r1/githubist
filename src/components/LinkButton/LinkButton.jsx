// @flow

import React, { type Node } from 'react';
import { Link, type LocationShape } from 'react-router-dom';
import cx from 'classnames';

import styles from './LinkButton.scss';

type Props = {
  to: string | LocationShape,
  children: Node,
  primary?: boolean,
  secondary?: boolean,
  small?: boolean,
  className?: string,
};

const LinkButton = ({ children, to, primary, secondary, small, className }: Props) => {
  const classNames = cx(
    styles.button,
    {
      [styles.primary]: primary,
      [styles.secondary]: secondary,
      [styles.small]: small,
    },
    className,
  );

  return (
    <Link to={to} className={classNames}>
      {children}
    </Link>
  );
};

LinkButton.defaultProps = {
  primary: false,
  secondary: false,
  small: false,
  className: '',
};

export default LinkButton;
