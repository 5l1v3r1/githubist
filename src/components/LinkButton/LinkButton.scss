@import '../../theme/theme';

.button {
  display: inline-flex;
  align-items: center;
  padding: px-to-rem(13px) px-to-rem(26px);
  border: 0;
  border-radius: px-to-rem(2px);
  cursor: pointer;
  font-size: px-to-rem(14px);
  letter-spacing: 0.05556rem;
  text-decoration: none;
  transition-duration: 0.3s;
  transition-property: background-color, color, border-color;
}

.small {
  padding: px-to-rem(7px) px-to-rem(13px);
  font-size: px-to-rem(12px);
}

.primary {
  background-color: $primary-color;
  color: $white;

  &:hover {
    background-color: darken($primary-color, 5%);
  }
}

.secondary {
  background-color: $secondary-color;
  color: $white;

  &:hover {
    background-color: darken($secondary-color, 5%);
  }
}
