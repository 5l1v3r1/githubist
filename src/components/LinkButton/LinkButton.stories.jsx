// @flow

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { storiesOf } from '@storybook/react';

import LinkButton from './LinkButton';

storiesOf('LinkButton', module)
  .add('primary', () => (
    <MemoryRouter>
      <LinkButton to="/" primary>
        Primary Button
      </LinkButton>
    </MemoryRouter>
  ))
  .add('secondary', () => (
    <MemoryRouter>
      <LinkButton to="/" secondary>
        Secondary Button
      </LinkButton>
    </MemoryRouter>
  ))
  .add('small', () => (
    <MemoryRouter>
      <LinkButton to="/" primary small>
        Small Button
      </LinkButton>
    </MemoryRouter>
  ));
