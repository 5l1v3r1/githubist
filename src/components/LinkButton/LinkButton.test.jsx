// @flow

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import LinkButton from './LinkButton';

it('renders correctly primary button', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkButton to="/" primary>
        Button
      </LinkButton>
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly secondary button', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkButton to="/" secondary>
        Button
      </LinkButton>
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly small button', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkButton to="/" small>
        Button
      </LinkButton>
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('passes custom classnames', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkButton to="/" className="testClassName">
        Button
      </LinkButton>
    </MemoryRouter>,
  );

  expect(wrapper.find('a').hasClass('testClassName')).toBe(true);
});
