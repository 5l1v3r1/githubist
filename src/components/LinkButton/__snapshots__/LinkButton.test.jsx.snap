// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly primary button 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkButton
      className=""
      primary={true}
      secondary={false}
      small={false}
      to="/"
    >
      <Link
        className=""
        replace={false}
        to="/"
      >
        <a
          className=""
          href="/"
          onClick={[Function]}
        >
          Button
        </a>
      </Link>
    </LinkButton>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly secondary button 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkButton
      className=""
      primary={false}
      secondary={true}
      small={false}
      to="/"
    >
      <Link
        className=""
        replace={false}
        to="/"
      >
        <a
          className=""
          href="/"
          onClick={[Function]}
        >
          Button
        </a>
      </Link>
    </LinkButton>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly small button 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkButton
      className=""
      primary={false}
      secondary={false}
      small={true}
      to="/"
    >
      <Link
        className="undefined"
        replace={false}
        to="/"
      >
        <a
          className="undefined"
          href="/"
          onClick={[Function]}
        >
          Button
        </a>
      </Link>
    </LinkButton>
  </Router>
</MemoryRouter>
`;
